modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java

```diff
@@ -332,6 +339,162 @@ public void testTransactions() throws Exception {
+
+        assertEquals(1, res.size());
+
+        List tbl = res.get(0);
+
+        int cacheId = cacheId("SQL_PUBLIC_T1");
+        String cacheName = "SQL_PUBLIC_T1";
+
+        assertEquals("T1", tbl.get(0)); //TABLE_NAME
+        assertEquals(DFLT_SCHEMA, tbl.get(1)); //SCHEMA_NAME
+        assertEquals(cacheName, tbl.get(2)); //CACHE_NAME
+        assertEquals(cacheId, tbl.get(3)); //CACHE_ID
+        assertNull(tbl.get(4)); //AFFINITY_KEY_COLUMN
+        assertEquals("ID", tbl.get(5)); //KEY_ALIAS
+        assertNull(tbl.get(6)); //VALUE_ALIAS
+        assertEquals("java.lang.Long", tbl.get(7)); //KEY_TYPE_NAME
+        assertNotNull(tbl.get(8)); //VALUE_TYPE_NAME
+
+        execute(ignite, "CREATE TABLE T2(ID LONG PRIMARY KEY, NAME VARCHAR)");
+
+        assertEquals(2, execute(ignite, "SELECT * FROM SYS.TABLES").size());
+
+        execute(ignite, "DROP TABLE T1");
+        execute(ignite, "DROP TABLE T2");
+
+        assertTrue(execute(ignite, "SELECT * FROM SYS.TABLES").isEmpty());
+    }
+
+    /** */
+    @Test
+    public void testTableColumns() throws Exception {
+        assertTrue(execute(ignite, "SELECT * FROM SYS.TABLE_COLUMNS").isEmpty());
+
+        execute(ignite, "CREATE TABLE T1(ID LONG PRIMARY KEY, NAME VARCHAR(40))");
+
+        Set<?> actCols = execute(ignite, "SELECT * FROM SYS.TABLE_COLUMNS")
+            .stream()
+            .map(l -> l.get(0))
+            .collect(Collectors.toSet());
+
+        assertEquals(new HashSet<>(asList("ID", "NAME", "_KEY", "_VAL")), actCols);
+
+        execute(ignite, "CREATE TABLE T2(ID LONG PRIMARY KEY, NAME VARCHAR(50))");
+
+        List<List<?>> expRes = asList(
+            asList("ID", "T1", "PUBLIC", false, false, "null", true, true, -1, -1, Long.class.getName()),
+            asList("NAME", "T1", "PUBLIC", false, false, "null", true, false, 40, -1, String.class.getName()),
+            asList("_KEY", "T1", "PUBLIC", true, false, null, false, true, -1, -1, null),
+            asList("_VAL", "T1", "PUBLIC", false, false, null, true, false, -1, -1, null),
+            asList("ID", "T2", "PUBLIC", false, false, "null", true, true, -1, -1, Long.class.getName()),
+            asList("NAME", "T2", "PUBLIC", false, false, "null", true, false, 50, -1, String.class.getName()),
+            asList("_KEY", "T2", "PUBLIC", true, false, null, false, true, -1, -1, null),
+            asList("_VAL", "T2", "PUBLIC", false, false, null, true, false, -1, -1, null)
+        );
+
+        List<List<?>> res = execute(ignite, "SELECT * FROM SYS.TABLE_COLUMNS ORDER BY TABLE_NAME, COLUMN_NAME");
+
+        assertEquals(expRes, res);
+
+        execute(ignite, "DROP TABLE T1");
+        execute(ignite, "DROP TABLE T2");
+
+        assertTrue(execute(ignite, "SELECT * FROM SYS.TABLE_COLUMNS").isEmpty());
+    }
+
+    /** */
+    @Test
+    public void testViewColumns() throws Exception {
+        execute(ignite, "SELECT * FROM SYS.VIEW_COLUMNS");
```

Comment:
What are we testing here?

Reply:
We fetch all the data from the view and check that it succeed